tests/integration/CartDiscount/CartDiscountCreateRequestTest.php

```diff
@@ -6,73 +6,52 @@
-use Commercetools\Core\Model\Common\Money;
-use Commercetools\Core\Model\Common\MoneyCollection;
-use Commercetools\Core\Request\CartDiscounts\CartDiscountCreateRequest;
-use Commercetools\Core\Request\CartDiscounts\CartDiscountDeleteRequest;
 
 class CartDiscountCreateRequestTest extends ApiTestCase
 {
-    /**
-     * @return CartDiscountDraft
-     */
-    protected function getDraft()
-    {
-        $draft = CartDiscountDraft::ofNameValuePredicateTargetOrderActiveAndDiscountCode(
-            LocalizedString::ofLangAndText('en', 'test-' . $this->getTestRun() . '-discount'),
-            CartDiscountValue::of()->setType('absolute')->setMoney(
-                MoneyCollection::of()->add(Money::ofCurrencyAndAmount('EUR', 100))
-            ),
-            '1=1',
-            CartDiscountTarget::of()->setType('lineItems')->setPredicate('1=1'),
-            '0.9' . trim((string)mt_rand(1, TestHelper::RAND_MAX), '0'),
-            true,
-            false
-        )->setKey('foo');
-
-        return $draft;
-    }
-
-    protected function createCartDiscount(CartDiscountDraft $draft)
-    {
-        $request = CartDiscountCreateRequest::ofDraft($draft);
-        $response = $request->executeWithClient($this->getClient());
-        $cartDiscount = $request->mapResponse($response);
-
-        $this->cleanupRequests[] = CartDiscountDeleteRequest::ofIdAndVersion(
-            $cartDiscount->getId(),
-            $cartDiscount->getVersion()
-        );
-
-        return $cartDiscount;
-    }
-
-
     public function testCreate()
     {
-        $draft = $this->getDraft();
-        $cartDiscount = $this->createCartDiscount($draft);
-        $this->assertSame($draft->getName()->en, $cartDiscount->getName()->en);
-        $this->assertSame(
-            $draft->getValue()->getMoney()->current()->getCentAmount(),
-            $cartDiscount->getValue()->getMoney()->current()->getCentAmount()
-        );
-        $this->assertSame(
-            $draft->getValue()->getMoney()->current()->getCurrencyCode(),
-            $cartDiscount->getValue()->getMoney()->current()->getCurrencyCode()
+        $client = $this->getApiClient();
+
+        CartDiscountFixture::withDraftCartDiscount(
+            $client,
+            function (CartDiscountDraft $draft) {
+                return $draft->setName(LocalizedString::ofLangAndText('en', 'myCategory'));
+            },
+            function (CartDiscount $cartDiscount) use ($client) {
+                $request = RequestBuilder::of()->cartDiscounts()->query()
+                    ->where('name(en=:name)', ['name' => $cartDiscount->getName()->en]);
+                $response = $client->execute($request);
+                $result = $request->mapFromResponse($response);
+
+                $this->assertSame($cartDiscount->getName()->en, $result->current()->getName()->en);
+                $this->assertSame(
+                    $result->current()->getValue()->getMoney()->current()->getCentAmount(),
```

Comment:
You are testing for exactly the same value

```suggestion
                    $cartDiscount->getValue()->getMoney()->current()->getCentAmount(),
```

This should also be changed at the other assertions